### PR TITLE
Resend cacert after etcdbackup restore

### PR DESCRIPTION
**Pull request by moelsayed**
#17521

### CODE REVIEW

File: package/kubectl-shell.sh [hunk 1]
```diff
@@ -14,8 +15,13 @@ if [ -z "${cluster}" ]; then
     exit 1
 fi
 
+if [ -z "${cacerts}" ]; then
+    echo No cacerts provided
```
Comment (alena1108) [inline package/kubectl-shell.sh]: @moelsayed could you confirm it won't break for clusters on upgraded setups?

File: package/shell-setup.sh [hunk 2]
```diff
@@ -15,9 +16,13 @@ SERVER="${CATTLE_SERVER}/k8s/clusters/${cluster}"
 if [ -f /etc/kubernetes/ssl/certs/serverca ]; then
     cp /etc/kubernetes/ssl/certs/serverca .kube/certs/ca.crt
     chmod 666 .kube/certs/ca.crt
-    if ! curl -s $SERVER > /dev/null; then
-        CERT="    certificate-authority: /nonexistent/.kube/certs/ca.crt"
-    fi
+else
+    echo ${cacerts} | base64 -d > .kube/certs/ca.crt
```
Comment (ibuildthecloud) [inline package/shell-setup.sh]: quote again

File: package/shell-setup.sh [hunk 3]
```diff
@@ -15,9 +16,13 @@ SERVER="${CATTLE_SERVER}/k8s/clusters/${cluster}"
 if [ -f /etc/kubernetes/ssl/certs/serverca ]; then
     cp /etc/kubernetes/ssl/certs/serverca .kube/certs/ca.crt
     chmod 666 .kube/certs/ca.crt
-    if ! curl -s $SERVER > /dev/null; then
-        CERT="    certificate-authority: /nonexistent/.kube/certs/ca.crt"
-    fi
+else
```
Comment (ibuildthecloud) [inline package/shell-setup.sh]: to keep old behavior, do `elif [ -n "${cacerts}" ]` here

File: package/shell-setup.sh [hunk 4]
```diff
@@ -15,9 +16,13 @@ SERVER="${CATTLE_SERVER}/k8s/clusters/${cluster}"
...
+    echo ${cacerts} | base64 -d > .kube/certs/ca.crt
+    chmod 666 .kube/certs/ca.crt
+fi
+
+if ! curl -s $SERVER > /dev/null; then
```
Comment (ibuildthecloud) [inline package/shell-setup.sh]: to keep old behavior, do `if [ -f /nonexistent/.kube/certs/ca.crt ] && ! curl -s $SERVER > /dev/null` here

File: package/kubectl-shell.sh (hunk 1 above)
Comment (moelsayed) [reply package/kubectl-shell.sh]: Upgrading and downgrading rancher server will trigger a node agent update. I have tested upgrade from v2.1.6 to master and rollback to v2.1.6 again with no issues.

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**